### PR TITLE
Improved docs in pxd's

### PR DESCRIPTION
**Pull request by rahul799**
# Description

This PR is intended to add function descriptions in pxd files.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

File: src/pyOpenMS/pxds/MSExperiment.pxd [hunk 1]
```diff
@@ -74,13 +74,13 @@ cdef extern from "<OpenMS/KERNEL/MSExperiment.h>" namespace "OpenMS":
         void reserveSpaceSpectra(Size s) nogil except +
         void reserveSpaceChromatograms(Size s) nogil except +
 
-        double getMinMZ() nogil except +
-        double getMaxMZ() nogil except +
-        double getMinRT() nogil except +
-        double getMaxRT() nogil except +
+        double getMinMZ() nogil except + # wrap-doc:returns the minimal m/z value
+        double getMaxMZ() nogil except + # wrap-doc:returns the maximal m/z value
+        double getMinRT() nogil except + # wrap-doc:eturns the minimal retention time value
```
Comment (jpfeuffer) [inline src/pyOpenMS/pxds/MSExperiment.pxd:79]:
```suggestion
        double getMinRT() nogil except + # wrap-doc:returns the minimal retention time value
```

File: src/pyOpenMS/pxds/MSExperiment.pxd [hunk 2]
```diff
@@ -74,13 +74,13 @@ cdef extern from "<OpenMS/KERNEL/MSExperiment.h>" namespace "OpenMS":
...
+        double getMaxRT() nogil except + # wrap-doc:eturns the maximal retention time value
```
Comment (jpfeuffer) [inline src/pyOpenMS/pxds/MSExperiment.pxd:80]:
```suggestion
        double getMaxRT() nogil except + # wrap-doc:returns the maximal retention time value
```

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: src/pyOpenMS/pxds/MSExperiment.pxd (hunk 1 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/MSExperiment.pxd]: Fixed in the latetst commit

File: src/pyOpenMS/pxds/MSExperiment.pxd (hunk 2 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/MSExperiment.pxd]: Fixed in the latest commit

**Comment by rahul799** [comment]
@jpfeuffer This PR is ready for the final review.

File: src/pyOpenMS/pxds/AASequence.pxd [hunk 3]
```diff
@@ -36,89 +36,89 @@ cdef extern from "<OpenMS/CHEMISTRY/AASequence.h>" namespace "OpenMS":
         String toUnmodifiedString() nogil except +
 
         # returns the peptide as string without any modifications
-        String toUniModString() nogil except +
+        String toUniModString() nogil except + # wrap-doc:returns the peptide as string with UniMod-style modifications embedded in brackets
 
         String toBracketString() nogil except + #wrap-doc:create a TPP compatible string of the modified sequence using bracket notation. Uses integer mass by default.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/AASequence.pxd:41]: This is a bit nitpicking, but maybe we could standardize if we use `# warp-doc` or `#wrap_doc` (at least per-file basis)

**Comment by oliveralka** [comment]
What do you think, should capital letters be used at the beginning of the sentence, when using `# wrap-doc`?
e.g.
`# wrap-doc:returns the peptide as string with UniMod-style modifications embedded in brackets`
or
`# wrap-doc:Returns the peptide as string with UniMod-style modifications embedded in brackets`

File: src/pyOpenMS/pxds/AASequence.pxd (hunk 3 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/AASequence.pxd:41]: Hey @oliveralka,
Yes, I am using `# warp-doc` everywhere.

**Comment by rahul799** [comment]
Yes, the first letter must be capitalized. I will make the changes.

File: src/pyOpenMS/pxds/AASequence.pxd (hunk 3 above)
Comment (oliveralka) [reply src/pyOpenMS/pxds/AASequence.pxd:41]: Could you please change it, if you see that it is done differently in the same file e.g. in line 41.

Comment (rahul799) [reply src/pyOpenMS/pxds/AASequence.pxd:41]: Sure.

File: src/pyOpenMS/pxds/AASequence.pxd [hunk 4]
```diff
@@ -36,89 +36,89 @@ cdef extern from "<OpenMS/CHEMISTRY/AASequence.h>" namespace "OpenMS":
...
 
         # returns the peptide as string without any modifications
-        String toUniModString() nogil except +
+        String toUniModString() nogil except + # wrap-doc:Returns the peptide as string with UniMod-style modifications embedded in brackets
 
-        String toBracketString() nogil except + #wrap-doc:create a TPP compatible string of the modified sequence using bracket notation. Uses integer mass by default.
-        String toBracketString(bool integer_mass) nogil except + #wrap-doc:create a TPP compatible string of the modified sequence using bracket notation.
-        String toBracketString(bool integer_mass, bool mass_delta) nogil except + #wrap-doc:create a TPP compatible string of the modified sequence using bracket notation.
-        String toBracketString(bool integer_mass, bool mass_delta, libcpp_vector[String] fixed_modifications) nogil except + #wrap-doc:create a TPP compatible string of the modified sequence using bracket notation.
+        String toBracketString() nogil except + #wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation. Uses integer mass by default.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/AASequence.pxd:41]:
```suggestion
        String toBracketString() nogil except + # wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation. Uses integer mass by default.
```

File: src/pyOpenMS/pxds/AASequence.pxd [hunk 5]
```diff
@@ -36,89 +36,89 @@ cdef extern from "<OpenMS/CHEMISTRY/AASequence.h>" namespace "OpenMS":
...
+        String toBracketString(bool integer_mass) nogil except + #wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/AASequence.pxd:42]:
```suggestion
        String toBracketString(bool integer_mass) nogil except + # wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation.
```

File: src/pyOpenMS/pxds/AASequence.pxd [hunk 6]
```diff
@@ -36,89 +36,89 @@ cdef extern from "<OpenMS/CHEMISTRY/AASequence.h>" namespace "OpenMS":
...
+        String toBracketString(bool integer_mass, bool mass_delta) nogil except + #wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation.
+        String toBracketString(bool integer_mass, bool mass_delta, libcpp_vector[String] fixed_modifications) nogil except + #wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/AASequence.pxd:44]:
```suggestion
        String toBracketString(bool integer_mass, bool mass_delta, libcpp_vector[String] fixed_modifications) nogil except + # wrap-doc:Create a TPP compatible string of the modified sequence using bracket notation.
```

**Review by oliveralka — COMMENTED** [review]
Looks good, with only a few minor changes (to fit the consensus).

**Comment by oliveralka** [comment]
You can use the files tab, to add the `suggestion to batch` and then commit all of them at once.

**Comment by rahul799** [comment]
Yes sure. @oliveralka actually I will fix the spacing issues at the end. So I think we can skip this for now and merge the PR. What do you think?

**Comment by oliveralka** [comment]
Yes, alright, that makes sense.

**Review by oliveralka — APPROVED** [review]
Looks good!

File: src/pyOpenMS/pxds/AbsoluteQuantitation.pxd [hunk 7]
```diff
@@ -37,7 +37,7 @@ cdef extern from "<OpenMS/ANALYSIS/QUANTITATION/AbsoluteQuantitation.h>" namespa
 
         # libcpp_map[ String, AbsoluteQuantitationMethod ] getQuantMethodsAsMap() nogil except +
 
-        double calculateBias(double actual_concentration, double calculated_concentration) nogil except +
+        double calculateBias(double actual_concentration, double calculated_concentration) nogil except + # wrap-doc:This function calculates the bias of the calibration.
```
Comment (jpfeuffer) [inline src/pyOpenMS/pxds/AbsoluteQuantitation.pxd:40]: What about dots now?

File: src/pyOpenMS/pxds/AhoCorasickAmbiguous.pxd [hunk 8]
```diff
@@ -12,6 +12,6 @@ cdef extern from "<OpenMS/ANALYSIS/ID/AhoCorasickAmbiguous.h>" namespace "OpenMS
         AhoCorasickAmbiguous(const String & protein_sequence) nogil except +
         void setProtein(const String & protein_sequence) nogil except +
         # bool findNext(const FuzzyACPattern & pattern) nogil except +
-        Size getHitDBIndex() nogil except +
-        Int getHitProteinPosition() nogil except +
+        Size getHitDBIndex() nogil except + # wrap-doc:Get index of hit into peptide database of the pattern.
+        Int getHitProteinPosition() nogil except + # wrap-doc:Offset into protein sequence where hit was found.
```
Comment (jpfeuffer) [inline src/pyOpenMS/pxds/AhoCorasickAmbiguous.pxd:16]: Dot

File: src/pyOpenMS/pxds/AhoCorasickAmbiguous.pxd [hunk 9]
```diff
@@ -12,6 +12,6 @@ cdef extern from "<OpenMS/ANALYSIS/ID/AhoCorasickAmbiguous.h>" namespace "OpenMS
         AhoCorasickAmbiguous(const String & protein_sequence) nogil except +
         void setProtein(const String & protein_sequence) nogil except +
         # bool findNext(const FuzzyACPattern & pattern) nogil except +
-        Size getHitDBIndex() nogil except +
-        Int getHitProteinPosition() nogil except +
+        Size getHitDBIndex() nogil except + # wrap-doc:Get index of hit into peptide database of the pattern.
```
Comment (jpfeuffer) [inline src/pyOpenMS/pxds/AhoCorasickAmbiguous.pxd:15]: Dot

**Review by jpfeuffer — COMMENTED** [review]
I thought we wanted to harmonize punctuation?

File: src/pyOpenMS/pxds/AbsoluteQuantitation.pxd (hunk 7 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/AbsoluteQuantitation.pxd]: My bad. Resolved now.

**Comment by oliveralka** [comment]
rebuild jenkins

**Pull request merged**